docs/application/dotnet/guides/app-management/component-port.md

```diff
@@ -0,0 +1,466 @@
+                _port.RequestReceived += OnRequestReceived;
+                _thread = new Thread(new ThreadStart(OnThread));
+                _thread.Start();
+                return true;
+            }
+
+            public override void OnStartCommand(AppControl appControl, bool restarted)
+            {
+                Log.Info("OnStartCommand()");
+                base.OnStartCommand(appControl, restarted);
+            }
+
+            public override void OnDestroy()
+            {
+                Log.Info("OnDestroy()");
+                base.OnDestroy();
+
+                if (_port != null)
+                {
+                    _port.RequestReceived -= OnRequestReceived;
+                    _port.Dispose();
+                    _port = null;
+                }
+            }
+        }
+    }
+    ```
+
+3.  Set up the receiving thread of the component.
+
+    1. To have the receiving thread of the component wait for incoming requests, call `WaitForEvent()` of the `Tizen.Applications.ComponentBased.ComponentPort` class.
+    2. If `WaitForEvent()` is called in the main thread, then `WaitForEvent()` can not be called until the `Cancel()` is called.
+    3. `WaitForEvent()` starts the main loop to wait for events from other components. To avoid blocking the main thread, it is recommended to use the `Tizen.Applications.ComponentBased.ComponentTask` class.
+    4. To handle the access control, call `AddPrivilege()` of the `Tizen.Applications.ComponentBased.ComponentPort` class. If a client does not have permission, the request is rejected with the permission denied error.
+    5. To handle the received request, define and register an event handler for the `RequestReceived` event of the `Tizen.Applications.ComponentBased.ComponentPort` class as follows:
+
+        ```csharp
+        {
+            _port = new ComponentPort("ServerService");
+            _port.RequestReceived += OnRequestReceived;
+            _port.AddPrivilege("http://tizen.org/privilege/datasharing");
+            _thread = new Thread(new ThreadStart(OnThread));
+            _thread.Start();
+        }
+
+        private void OnThread()
+        {
+            Log.Info("OnTrhead()");
+            _port.WaitForEvent();
+        }
+
+        private void OnRequestedReceived(object sender, RequestEventArgs args)
+        {
+            Log.Info("Sender: + args.Sender);
+            var request = args.Request;
+            if (request.GetType() == typeof(Request))
+            {
+                var req = (Request)request;
+                Log.Info("Command: " + req.Command);
+                Log.Info("Sequence: " + req.Sequence);
+                Log.Info("Message: " + req.Message);
+
+                if (args.IsReplyRequested)
+                {
+                    args.Reply = new Response(req.Command, req.Sequence, 0);
+                }
+            }
+        }
+        ```
+
+4.  Send the request in the sending component as follows:
+
```

Comment:
@nawab-samsung, @hjhun, is this sequential? An ordered list is needed here?

Reply:
Yes. I think it's needed.